frontend-web/webclient/app/Applications/Jobs/View.tsx

```diff
@@ -827,11 +860,103 @@ function parseUpdatesForAccess(updates: JobUpdate[]): ParsedSshAccess | null {
     return null;
 }
 
+interface InterfaceTarget {
+    rank: number,
+    type: "WEB" | "VNC" | "SHELL";
+    target?: string,
+    port?: number,
+    link?: string,
+}
+
+async function findInterfaceTargets(job: Job, invokeCommand: InvokeCommand): Promise<InterfaceTarget[]> {
+    const result: InterfaceTarget[] = [];
+    const appType = getAppType(job);
+    const backendType = getBackend(job);
+    const support = job.status.resolvedSupport ?
+        (job.status.resolvedSupport! as ResolvedSupport<never, ComputeSupport>).support : undefined;
+
+    // Parse targets from job updates
+    for (let i = job.updates.length - 1; i >= 0; i--) {
+        const status = job.updates[i].status;
+        if (!status) continue;
+
+        const messages = status.split("\n");
+        for (const message of messages) {
+            if (message.startsWith("Target: ")) {
+                const parser = new SillyParser(message);
+                parser.consumeToken("Target: ");
+
+                const parsedTarget = JSON.parse(parser.remaining()) as InterfaceTarget;
+
+                const supportsInterface =
+                    (parsedTarget.type === "WEB" && isSupported(backendType, support, "web")) ||
+                    (parsedTarget.type === "VNC" && isSupported(backendType, support, "vnc"));
+
+                if (parsedTarget.type === "WEB" && supportsInterface && job.status.state === "RUNNING") {
+                    const requests: OpenInteractiveSessionRequest[] = Array(job.specification.replicas)
+                        .fill(0).map((_, i) => ({sessionType: "WEB", id: job.id, rank: i, target: parsedTarget.target, port: parsedTarget.port}));
+
+                    try {
+                        const sessionResult = await invokeCommand<BulkResponse<InteractiveSession>>(JobsApi.openInteractiveSession(bulkRequestOf(...requests)));
```

Comment:
Collect multiple requests in an array and then do a single call after the outer loop finishes. Include even the normal calls in this request, as this will also resolve the comment from Jonas.